src/System/refine.rs

```diff
@@ -4,11 +4,13 @@ use crate::ModelObjects::component::Transition;
 use crate::ModelObjects::max_bounds::MaxBounds;
 use crate::ModelObjects::statepair::StatePair;
 use crate::TransitionSystems::{LocationTuple, TransitionSystemPtr};
+use simple_error::bail;
+use std::error::Error;
 
 pub fn check_refinement(
     mut sys1: TransitionSystemPtr,
     mut sys2: TransitionSystemPtr,
-) -> Result<bool, String> {
+) -> Result<Result<bool, String>, Box<dyn Error>> {
```

Comment:
I don't like having Results with results inside here, why can we not just have a single result?